File: j8/pomodoro-timer/src/test/java/knowbag/pomodoro/state/prototype/StateTest.java
package knowbag.pomodoro.state.prototype;

import org.junit.Test;

/**
 * Created by feliperojas on 29/03/15.
 */
public class StateTest {

    @Test
    public void testWorkingState() {
        PomodoroFlow pomodoroFlow = new PomodoroFlow();
        pomodoroFlow.executeState();
        pomodoroFlow.executeState();
        pomodoroFlow.executeState();
        pomodoroFlow.executeState();
        pomodoroFlow.executeState();
        pomodoroFlow.executeState();
    }
}

File: scala/functional-state/src/test/scala-2.11/knowbag/functional/state/StateTest.scala
package knowbag.functional.state

import org.scalatest.{FlatSpec, Matchers}

/**
 * Created by feliperojas on 7/19/15.
 */
class StateTest extends FlatSpec with Matchers {

  "a state" should "have the value for the current state ant the transition to the next one" in {
    val state = State[String, Integer](state => (10, s"$state value"))
    val finalState = state.map(_ * 2).run("nice")
    println(finalState)
  }

}
